src/main/java/pl/piekoszek/gorskimatches/equation/EquationMathChecker.java

```diff
@@ -2,11 +2,36 @@
 
 import org.springframework.stereotype.Component;
 
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
 @Component
 class EquationMathChecker {
 
+    Map<String, Boolean> allEquations = new HashMap<>();
```

Comment:
I guess `listOfMathCorrEquations` should be stored as class field instead of this.